airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2221,10 +2217,16 @@ def execute(self, context: Any):
                             BigQueryTableLink.persist(**persist_kwargs)
 
         self.job_id = job.job_id
-        return job.job_id
+        # Wait for the job to complete
+        job.result(timeout=self.result_timeout, retry=self.result_retry)
+        self._handle_job_error(job)
+
+        return self.job_id
 
     def on_kill(self) -> None:
         if self.job_id and self.cancel_on_kill:
             self.hook.cancel_job(  # type: ignore[union-attr]
                 job_id=self.job_id, project_id=self.project_id, location=self.location
             )
+        else:
+            self.log.info(f'Skipping to cancel job: {self.project_id}:{self.location}.{self.job_id}')
```

Comment:
```suggestion
            self.log.info('Skipping to cancel job: %s:%s.%s', self.project_id, self.location, self.job_id)
```
We should [try to avoid f-strings in logging statements](https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/) and use %-format style instead.